### PR TITLE
test: Replace card component test with testing lib

### PR DESCRIPTION
**Pull request by LongHaoo**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

1. 将原来的测试文件改为 tsx格式，从而可以看到代码提示校验等
2. 更新了一个旧的Snapshot
3. 将原来的测试逻辑进行替换，符合 testing lib 的测试模式

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/card/__tests__/__snapshots__/demo-extend.test.ts.snap [hunk 1]
```diff
@@ -912,7 +912,7 @@ Array [
             <div>
               <div
                 class="ant-tabs-dropdown"
-                style="opacity:0"
+                style="opacity:0;pointer-events:none"
```
Comment (afc163) [inline components/card/__tests__/__snapshots__/demo-extend.test.ts.snap:915]: 应该不需要改这个。

Comment (afc163) [reply components/card/__tests__/__snapshots__/demo-extend.test.ts.snap:915]: 本地 node_modules 更新一下。

Comment (LongHaoo) [reply components/card/__tests__/__snapshots__/demo-extend.test.ts.snap:915]: master yarn 更新之后

Comment (afc163) [reply components/card/__tests__/__snapshots__/demo-extend.test.ts.snap:915]: 本地有 yarn.lock 吧，删掉再更新。

**Pull request merged**